contents/docs/how-posthog-works/clickhouse.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Clickhouse
+---
+
+ClickHouse is our main analytics backend.
+
+Instead of data being inserted directly into ClickHouse, it itself data from Kafka. This makes our ingestion pipeline more resilient towards outages.
+
+The following sections go more into depth in how this works exactly.
+
+### Non-sharded ClickHouse events ingestion
+
+```mermaid
+flowchart LR
+    classDef table fill:#f6e486,stroke:#ffc45d;
+    kafka_events["kafka_events table<br/>(Kafka table engine)"]:::table
+    eventsmv["events_mv table<br/>(Materialized view)"]:::table
+    events["events table<br/>(ReplacingMergeTree table engine)"]:::table
+
+    kafka_events -.clickhouse_events_json topic.- Kafka
+    eventsmv --reads from--> kafka_events
+    eventsmv --pushes data to--> events
+```
+
+#### `kafka_events` table
+
+`kafka_events` table is of [Kafka table engine](https://clickhouse.com/docs/en/engines/table-engines/integrations/kafka/)
+
+In essence it behaves as a kafka consumer group - reading from this table reads from the underlying kafka topic and advances the current offset.
```

Comment:
Unsure how to make this sentence simple, but:

```suggestion
Tables using this engine set up Kafka consumers that consume data on read queries to the table, advancing the offset for the consumer group in Kafka.
```

Cause currently it's slightly technically incorrect.